components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/device/grant/DeviceFlowGrant.java

```diff
@@ -0,0 +1,185 @@
+                        String authzUser = results.get(Constants.AUTHZ_USER).toString();
+                        String[] scopeSet = OAuth2Util.buildScopeArray(results.get(Constants.SCOPE).toString());
+                        this.setPropertiesForTokenGeneration(oAuthTokenReqMessageContext, tokenReq, scopeSet,
+                                authzUser);
+                    }
+                } else if (deviceStatus.equals(Constants.USED) || deviceStatus.equals(Constants.PENDING)) {
+                    Timestamp newPollTime = new Timestamp(date.getTime());
+                    if (isValidPollTime(newPollTime, results)) {
+                        DeviceFlowPersistenceFactory.getInstance().getDeviceFlowDAO()
+                                .setLastPollTime(DeviceCode, newPollTime);
+                        throw new IdentityOAuth2Exception(DeviceErrorCodes.SubDeviceErrorCodes.AUTHORIZATION_PENDING);
+                    } else {
+                        DeviceFlowPersistenceFactory.getInstance().getDeviceFlowDAO()
+                                .setLastPollTime(DeviceCode, newPollTime);
+                        throw new IdentityOAuth2Exception(DeviceErrorCodes.SubDeviceErrorCodes.SLOW_DOWN);
+                    }
+                }
+            }
+        }
+        return authStatus;
+    }
+
+    /**
+     * @param tokReqMsgCtx
+     * @return true
+     * @throws IdentityOAuth2Exception
+     */
+    public boolean authorizeAccessDelegation(OAuthTokenReqMessageContext tokReqMsgCtx)
+            throws IdentityOAuth2Exception {
+
+        return true;
+    }
+
+    /**
+     * @param tokReqMsgCtx
+     * @return true
+     * @throws IdentityOAuth2Exception
+     */
+    public boolean validateScope(OAuthTokenReqMessageContext tokReqMsgCtx)
+            throws IdentityOAuth2Exception {
+
+        // if we need to just ignore the scope verification
+        return true;
+    }
+
+    /**
+     * @param tokReqMsgCtx
+     * @param tokenReq
+     * @param scopes  scopes that will be stored against token
+     * @param authzUser authorized user
+     */
+    private void setPropertiesForTokenGeneration(OAuthTokenReqMessageContext tokReqMsgCtx,
+                                                 OAuth2AccessTokenReqDTO tokenReq, String[] scopes, String authzUser) {
+
+        tokReqMsgCtx.setAuthorizedUser(OAuth2Util.getUserFromUserName(authzUser));
+        tokReqMsgCtx.setScope(scopes);
+    }
+
+    @Override
+    public OAuth2AccessTokenRespDTO issue(OAuthTokenReqMessageContext tokReqMsgCtx) throws IdentityOAuth2Exception {
+
+        OAuth2AccessTokenRespDTO tokenRespDTO = super.issue(tokReqMsgCtx);
+        return tokenRespDTO;
+    }
+
+    /**
+     * This method use to check whether device code is expired or not
+     * @param results result map that contains values from database
+     * @param date time that request has came
+     * @return true or false
+     */
+    private boolean isValidaDeviceCode(HashMap results, Date date) {
```

Comment:
This method should be 'isValidDeviceCode' right?

Reply:
Yes akka

Reply:
Fix typo in this method name